apps/bfd-pipeline/bfd-pipeline-ccw-rif/src/main/java/gov/cms/bfd/pipeline/ccw/rif/load/RifLoader.java

```diff
@@ -333,6 +339,21 @@ public Integer getValue() {
         throw new RuntimeException(e);
       }
 
+      /*
+       * If any batch load tripped a fatal error, we should throw an exception
+       * on this thread to kill the pipeline. Exceptions thrown from the batch thread with the
+       * error are ignored/lost, so have this synchronized boolean signal to do this specifically on the main thread.
+       * We also can't use a passed-in error handler from up the execution chain or else the pipeline manager shutdown procedure
+       * waits for this thread to shut down, which is waiting on the above 72-hour timeout before it unblocks.
+       *
+       * FUTURE: If we wish to be more discerning about recoverable errors, we could return false here which will
+       * allow the pipeline to continue operating if an error is considered non-fatal (it also currently moves the failed files
+       * to "Done" instead of "Failed", which would need to be fixed.)
+       */
+      if (fatalFailure) {
+        throw new IllegalStateException("Fatal error during rif file processing.");
+      }
```

Comment:
This job needs to definitively return in order to shut down gracefully, so we still can't use the error handler here either. Instead, throw an exception and PipelineManager will trigger the job as failed and shut things down for us.